package com.flechazo.sakuraFabric.loot_modifier;

import com.google.common.collect.Lists;
import com.mojang.datafixers.util.Pair;
import com.mojang.serialization.*;
import it.unimi.dsi.fastutil.objects.ObjectArrayList;
import net.minecraft.world.item.Item;
import net.minecraft.world.item.ItemStack;
import net.minecraft.world.level.storage.loot.LootContext;
import net.minecraft.world.level.storage.loot.predicates.LootItemCondition;

import javax.annotation.Nonnull;
import java.util.List;

public class SeedsDrop {
    public static class SeedDropModifier extends LootModifier {
        public SeedDropModifier(LootItemCondition[] conditionsIn) {
            super(conditionsIn);
        }

        public static final Codec<SeedDropModifier> CODEC = Codec.of(
                new Encoder<SeedDropModifier>() {
                    @Override
                    public <T> DataResult<T> encode(SeedDropModifier seedDropModifier, DynamicOps<T> dynamicOps, T t) {
                        return null;
                    }
                },
                new Decoder<SeedDropModifier>() {
                    @Override
                    public <T> DataResult<Pair<SeedDropModifier, T>> decode(DynamicOps<T> dynamicOps, T t) {
                        return null;
                    }
                }
        );

        @Nonnull
        @Override
        protected ObjectArrayList<ItemStack> doApply(ObjectArrayList<ItemStack> generatedLoot, LootContext context) {
            List<Item> seeds = Lists.newArrayList(ItemRegistry.CABBAGE_SEEDS.get(), ItemRegistry.EGGPLANT_SEEDS.get(),
                    ItemRegistry.ONION_SEEDS.get(), ItemRegistry.RADISH_SEEDS.get(), ItemRegistry.TOMATO_SEEDS.get(),
                    ItemRegistry.RICE_SEEDS.get(), ItemRegistry.RAPESEEDS.get(), ItemRegistry.TARO.get(),
                    ItemRegistry.BUCKWHEAT.get(), ItemRegistry.SOYBEAN.get(), ItemRegistry.RED_BEAN.get());
            generatedLoot.add(new ItemStack(seeds.get((int) (Math.random() * seeds.size()))));
            return generatedLoot;
        }

        @Override
        public Codec<? extends IGlobalLootModifier> codec() {
            return SeedDropModifier.DIRECT_CODEC;
        }
    }


}